packages/agoric-cli/src/commands/psm.js

```diff
@@ -81,20 +62,50 @@ export const makePsmCommand = async logger => {
   `,
   );
 
-  const lookupPsmInstance = ([minted, anchor]) => {
-    const name = `psm-${minted}-${anchor}`;
-    const instance = agoricNames.instance[name];
-    if (!instance) {
-      logger.debug('known instances:', agoricNames.instance);
-      throw new Error(`Unknown instance ${name}`);
-    }
-    return instance;
+  const rpcTools = async () => {
+    const { vstorage, fromBoard, agoricNames } = await makeRpcUtils({ fetch });
```

Comment:
ditto with comment above. if we do this often maybe worth another argument to `makeRpcUtils` that takes a maker for more utils.
```
const rpcTools = () => return makeRpcUtils(
  {fetch}, 
  base => {
    lookupPsmInstance: ([minted, anchor]) => {
      const name = `psm-${minted}-${anchor}`;
      const instance = base.agoricNames.instance[name];
      if (!instance) {
        logger.debug('known instances:', agoricNames.instance);
        throw new Error(`Unknown instance ${name}`);
      }
      return instance;
    }
  ```

Reply:
I usually go for the objects as closures pattern...

```js
const makeThing = powers => {
  ...
  return harden({ method1, method2, ... });l
}
```

... and think about POLA while I'm at it. The bundles of rights that come to mind are:

  - an RPC client, including vstorage. it would know about network-config and ideally do failover
  - such a client that also has `fromBoard` state and knows about board marshalling
  - keyring access: just map names to addresses, no private key access
  - keyring+RPC: a sign-and-broadcast tool